legal-api/src/legal_api/resources/v2/business/business_filings/business_filings.py

```diff
@@ -596,10 +606,17 @@ def save_filing(  # pylint: disable=too-many-return-statements,too-many-branches
                 return None, None, {"message": f"{business_identifier} not found"}, HTTPStatus.NOT_FOUND
 
             if client_request.method == "PUT":
+                # Model is based on the nature of the business
+                business_model = AlternateName if business.is_alternate_name_entity else LegalEntity
+                # Whether to query over the alternate_name_id or legal_entity_id column in the Filings table
+                # That depends also on the nature (model) of the business
+                filing_entity_id = (
+                    Filing.alternate_name_id if business.is_alternate_name_entity else Filing.legal_entity_id
+                )
                 rv = (
-                    db.session.query(LegalEntity, Filing)
-                    .filter(LegalEntity.id == Filing.legal_entity_id)
-                    .filter(LegalEntity.identifier == business_identifier)
+                    db.session.query(business_model, Filing)
+                    .filter(business_model.id == filing_entity_id)
+                    .filter(business_model.identifier == business_identifier)
```

Comment:
think there is a problem with `.filter(business_model.id == filing_entity_id)`.  These are both columns and the filter requires a column to be compared against a value I believe.

Maybe we need something like

```python
rv = (
        db.session.query(business_model, Filing)
        .join(Filing, filing_entity_id == business_model.id)  # join Filing table based on the dynamic column
        .filter(business_model.identifier == business_identifier)
        .filter(Filing.id == filing_id)
        .one_or_none()
    )
```

Reply:
Argus, I've made this change and verified:
For Firms:
![202 put worksafter comment](https://github.com/bcgov/lear/assets/122301442/e9bef37c-ab47-4e7c-a0aa-19ac3c24cc15)

For a BEN, I tried doing a PUT by making an alteration:
![alteration PUT](https://github.com/bcgov/lear/assets/122301442/a63324fe-eb03-4f0e-86c4-703dfc153f57)